lib/server.js

```diff
@@ -350,62 +347,19 @@ class Server extends KarmaEventEmitter {
       })
     }
 
-    const webServerCloseTimeout = 3000
-    const disconnectBrowsers = (code) => {
-      const sockets = socketServer.sockets.sockets
-
-      Object.keys(sockets).forEach((id) => {
-        const socket = sockets[id]
-        socket.removeAllListeners('disconnect')
-        if (!socket.disconnected) {
-          process.nextTick(socket.disconnect.bind(socket))
-        }
-      })
-
-      this.emitExitAsync(code).catch((err) => {
-        this.log.error('Error while calling exit event listeners\n' + err.stack || err)
-        return 1
-      }).then((code) => {
-        socketServer.sockets.removeAllListeners()
-        socketServer.close()
-
-        let removeAllListenersDone = false
-        const removeAllListeners = () => {
-          if (removeAllListenersDone) {
-            return
-          }
-          removeAllListenersDone = true
-          webServer.removeAllListeners()
-          processWrapper.removeAllListeners()
-          done(code || 0)
-        }
-
-        const closeTimeout = setTimeout(removeAllListeners, webServerCloseTimeout)
-
-        webServer.close(() => {
-          clearTimeout(closeTimeout)
-          removeAllListeners()
-        })
-      })
-    }
-
-    processWrapper.on('SIGINT', () => disconnectBrowsers(process.exitCode))
-    processWrapper.on('SIGTERM', disconnectBrowsers)
-
-    const reportError = (error) => {
-      process.emit('infrastructure_error', error)
```

Comment:
As far as I can see this even has never been used by anyone and was probably added by mistake.

Reply:
Oops we use that internally. I need to find way to get the equivalent behavior

Reply:
I can also restore it in this PR. Which would you prefer? The only reason for removal was that I though that it is unused.

Reply:
I'll just go ahead and restore it to not block this PR. We can remove it the separate PR if needed.

Reply:
Done